pandas/_testing.py

```diff
@@ -1456,7 +1456,19 @@ def assert_series_equal(
             check_dtype=check_dtype,
             index_values=np.asarray(left.index),
         )
-    elif needs_i8_conversion(left.dtype) or needs_i8_conversion(right.dtype):
+    elif (
+        is_extension_array_dtype(left.dtype) and needs_i8_conversion(right.dtype)
+    ) or (is_extension_array_dtype(right.dtype) and needs_i8_conversion(left.dtype)):
+        # If we have an extension array and Datetimearray / TimedeltaArray, we still
+        # want to assert_extension_array_equal even though Datetimearray and
```

Comment:
There is a lot going on here.
I _suggest_ extracting a function checking for this condition, with a proper docstring.

Reply:
You may also want to reference original issue in the comment or docstring (#37609).

Reply:
Okay! Will do.